### ISSUE TITLE
Connection interrupted, reconnecting...

### ISSUE THREAD

**Issue by radeon79** [OpenWebNet-HA/MyHOME#31]
Hi, i have a myhome server1. The connection is ok and all objects of my_home is work but after 2-3 minutes of boot of home assistant i receive this error:

Logger: custom_components.myhome
Source: /usr/local/lib/python3.9/site-packages/OWNd/connection.py:584
Integration: MyHome (documentation, issues)
First occurred: 15:54:48 (2 occurrences)
Last logged: 16:01:25

Connection interrupted, reconnecting...

and 

Logger: custom_components.myhome
Source: /usr/local/lib/python3.9/site-packages/OWNd/connection.py:276
Integration: MyHome (documentation, issues)
First occurred: 16:01:25 (1 occurrences)
Last logged: 16:01:25

Event session connection refused, retrying in 1s.

in this case all components of myhome is not more working on home assistant.
How to fix this issue?

**Comment by radeon79** [comment]
I have already set to 1 number session but the error persist. Is impossible to use for me this plugin. Is work for 2-3 minutes only.

**Comment by anotherjulien** [comment]
I honestly don't know what to tell you.
Few people have reported similar issues, but they are not limited to this plugin. Any software they use to try to interact with their gateway crashes ever 2 or 3 minutes.
Some have found issues with their network (IGMP snooping settings on their switch if I recall correctly), but some just never found anything obvious.
In some cases I must suppose the gateway is actually defective, but I don't know for sure.

**Comment by radeon79** [comment]
hi, no the gateway is ok. For over two years I have been using sdomotica and it works perfectly. Even now if I use sdomotica it works !!
If I block sdomotica and use your plugin, everything works perfectly but only for 2-3 minutes then that message appears and does not work anymore until restart

**Comment by mougeat** [comment]
Hi, I have the same problem. I have changed my gateway but I still have the problem. I have "solved it" by reloading the integration every 45s. You can try it manually once you have  lost the connection. 

In Configuration --> Integrations -->look for your gateway and click on the 3 dot and reload

If after that you can pilot again your lamps and switches, this could be your solution with an automation.

**Comment by anotherjulien** [comment]
What? you replaced your gateway and the problem continued exactly the same?
You replaced it with the same model?

**Comment by mougeat** [comment]
Yes I replaced it with the same model. I already have the problem but also with other applications (atMyHome)

My gateway disconnect from the network

**Comment by anotherjulien** [comment]
Then I would assume the problem must be environmental somehow.
Would you be able to provide some basic network layout? What's connected where and how?

**Comment by harisma-git** [comment]
I have the same problem. Every 5-6 days i see in the logs the message  Connection interrupted, reconnecting.... The solution is to reload the integration. The problem is that entities seems ok and not unavailable (of course nothing happens if you toggle a switch). If it is possible to go to unavavailble i can make an automation to restart the integration.

**Comment by radeon79** [comment]
the refresh is work but this is a trick, not a solution!!!

**Comment by anotherjulien** [comment]
But you have a solution, don't you? Just continue using Sdomotica :-/

**Comment by radeon79** [comment]
Of course no problem with sdomotica but I wold like to remove the raspberry and use only intel nuc, then I would like to solve this problem. Regards

**Comment by nicolacavallo84** [comment]
After a short time, the Myhome integration no longer works for me. To restore operation I have to restart homeassistant, but the problem reappears soon after.

**Comment by GreenGrassBlueOcean** [comment]
Thank you for reporting this issue.

This issue has been resolved in the modernized **v2 architecture** developed under the new **[OpenWebNet-HA](https://github.com/OpenWebNet-HA)** community organization:
- Addressed in **[Pull Request #232 (`v2-phase1-architecture`)](https://github.com/OpenWebNet-HA/MyHOME/pull/232)** and merged core commits.
- Resolved by declarative hardware profiles, token-bucket queue pacing, connection watchdog hardening, and canonical entity registry migration.

If you would like to test, please download or switch to the `v2-phase1-architecture` branch in HACS. If you encounter any remaining issues on this build, please feel free to report them in [PR #232](https://github.com/OpenWebNet-HA/MyHOME/pull/232).
